test/BMTFile.ts

```diff
@@ -0,0 +1,151 @@
+import { Chunk, fileInclusionProofBottomUp, getSpanValue, makeChunkedFile, Utils } from '@fairdatasociety/bmt-js'
+import { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { BMTFile } from '../typechain'
+import FS from 'fs'
+import path from 'path'
+import { hexlify } from 'ethers/lib/utils'
+
+describe('file', () => {
+  let bosBytes: Uint8Array
+  let carrierChunkFileBytes: Uint8Array
+  let bmtlib: BMTFile
+  const SEGMENT_SIZE = 32
+  before(async () => {
+    const BMT = await ethers.getContractFactory('BMTFile')
+    bmtlib = await BMT.deploy()
+    await bmtlib.deployed()
+
+    bosBytes = Uint8Array.from(FS.readFileSync(path.join(__dirname, 'test-files', 'The-Book-of-Swarm.pdf')))
+    carrierChunkFileBytes = Uint8Array.from(FS.readFileSync(path.join(__dirname, 'test-files', 'carrier-chunk-blob')))
+  })
+
+  it('should find BMT position of the payload segment index', async () => {
+    // edge case - carrier chunk
+    const fileBytes = carrierChunkFileBytes
+    const chunkedFile = makeChunkedFile(fileBytes)
+    const tree = chunkedFile.bmt()
+    const leafChunks = chunkedFile.leafChunks()
+    // check whether the last chunk is not present in the BMT tree 0 level -> carrierChunk
+    expect(tree[0].length).eq(leafChunks.length - 1)
+    const carrierChunk = leafChunks.pop() as Chunk
+    const segmentIndex = Math.floor((fileBytes.length - 1) / 32) // last segment index as well
+    const lastChunkIndex = Math.floor((fileBytes.length - 1) / 4096)
+    const segmentIdInTree = await bmtlib.getBmtIndexOfSegment(segmentIndex, lastChunkIndex)
+    expect(segmentIdInTree[1]).eq(1)
+    expect(segmentIdInTree[0]).eq(1)
+    const chunkIndex = segmentIdInTree[0].toNumber()
+    const level = segmentIdInTree[1].toNumber()
+    expect(hexlify(tree[level][chunkIndex].address())).eq(hexlify(carrierChunk.address()))
+  })
+
+  it('should collect the required segments for inclusion proof', async () => {
+    const fileBytes = carrierChunkFileBytes
+    const chunkedFile = makeChunkedFile(fileBytes)
+    const fileHash = chunkedFile.address()
+    // segment to prove
+    const lastSegmentIndex = Math.floor((fileBytes.length - 1) / 32)
+    // check segment array length for carrierChunk inclusion proof
+    const proofChunks = fileInclusionProofBottomUp(chunkedFile, lastSegmentIndex)
+    expect(proofChunks.length).eq(2) // 1 level is skipped because the segment was in a carrierChunk
+
+    /** Gives back the file hash calculated from the inclusion proof method */
+    const testGetFileHash = async (segmentIndex: number): Promise<string> => {
+      const proofChunks = fileInclusionProofBottomUp(chunkedFile, segmentIndex)
+      let proveSegment = fileBytes.slice(segmentIndex * SEGMENT_SIZE, segmentIndex * SEGMENT_SIZE + SEGMENT_SIZE)
+      // padding
+      proveSegment = new Uint8Array([...proveSegment, ...new Uint8Array(SEGMENT_SIZE - proveSegment.length)])
```

Comment:
why is it needed?

Reply:
tried to replace test 1 and 2 without padding code but left it. This is as is , taken from bmt.js.

Reply:
aa yeah, I see now... well, I haven't commented this test code unfortunately for myself, but the padding is needed because the sliced bytearray won't result in a full segment without it.